mllib/src/main/scala/org/apache/spark/mllib/stat/test/ChiSqTest.scala

```diff
@@ -109,7 +109,9 @@ private[stat] object ChiSqTest extends Logging {
           }
           i += 1
           distinctLabels += label
-          features.toArray.view.zipWithIndex.slice(startCol, endCol).map { case (feature, col) =>
+          val featureArray = features.toArray
+          (startCol until endCol).map { col =>
+            val feature = featureArray(col)
```

Comment:
@jkbradley I changed it to invoking toBreeze directly. I ran some test on local.

```
    val sv = new SparseVector(100000, Array(1), Array(2.5))
    var t = 0.0

    {
      val ts = System.nanoTime()
      val arr = sv.toArray
      for(i <- 0 to 10000){
        t = arr(i)
      }
      println(System.nanoTime() - ts)
    }

    {
      val ts = System.nanoTime()
      for(i <- 0 to 10000){
        t = sv(i)
      }
      println(System.nanoTime() - ts)
    }

    {
      val ts = System.nanoTime()
      val brz = sv.toBreeze
      for(i <- 0 to 10000){
        t = brz(i)
      }
      println(System.nanoTime() - ts)
    }
```

gives 
3405342
520367839
4870954

Let me know how do you think of it.